core/cloudflow-spark-testkit/src/main/scala/cloudflow/spark/testkit/SparkStreamletTestkit.scala

```diff
@@ -221,8 +221,12 @@ final case class SparkStreamletTestkit(session: SparkSession, config: Config = C
       sparkStreamlet: SparkStreamlet,
       duration: Duration
   ): Unit = {
+    val t0 = System.currentTimeMillis()
     val queryExecution = sparkStreamlet.setContext(ctx).run(ctx.config)
-    session.streams.awaitAnyTermination(duration.toMillis)
+    while (session.streams.active.nonEmpty && (System.currentTimeMillis()-t0)<duration.toMillis ) {
+      session.streams.awaitAnyTermination(duration.toMillis)
```

Comment:
Another options would be to call `processAllAvailable` and also be more strict: `session.streams.active.foreach(x => x.awaitTermination())` if any query fails then we should fail immediately.

Reply:
Interesting idea. I guess this is a tricky situation.  `streams.active` will be evaluated only once and it's possible that after we wait for the first query to terminate, other queries, previously considered active, are already terminated.
Also, total runtime will potentially be `#queries x timeout` in the worst case.

Check for active queries each time seems a better option to me. What do you think?

Reply:
With two streams both should finish fat, if not the test is not tuned anyway. It depends.